testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDeleteCompatibility.java

```diff
@@ -535,7 +537,8 @@ public void testDeletionVectorsAbsolutePath()
 
             List<Row> expected = ImmutableList.of(row(2, 22), row(4, 44));
             assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName)).contains(expected);
-            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName)).contains(expected);
+            assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
+                    .hasMessageContaining("Unsupported storage type for deletion vector: p");
```

Comment:
💪

Reply:
BTW this test is overly complicated IMO.

do we have a test exercising the deletion vectors but without SHALLOW CLONE?
Or is the CLONE operation the moment where deletion vectors are being converted  to `p` form?

- there should be a simpler test, not involving SHALLOW CLONE
- OR, there should be onTrino read from the baseTableName before the clone is created.

Reply:
after `DELETE` storageType of the deletion vector is `u`  meaning it has a path relative to table's directory

```
{"add":{"path":"part-00000-ec4adcbb-8616-4d93-ab3b-1343e49e2d9e-c000.snappy.parquet","partitionValues":{},"size":690,"modificationTime":1712605703000,"dataChange":true,"stats":"{\"numRecords\":4,\"minValues\":{\"a\":1,\"b\":11},\"maxValues\":{\"a\":4,\"b\":44},\"nullCount\":{\"a\":0,\"b\":0},\"tightBounds\":false}","deletionVector":{"storageType":"u","pathOrInlineDv":"5)W-R!w$fPLzXnwyv5Km","offset":1,"sizeInBytes":36,"cardinality":2}}}
```

after `SHALLOW CLONE` the active files of the table will be referenced with absolute path. Their eventual deletion vectors will have storageType set to `p` meaning that they have an absolute path.

```
{"add":{"path":"s3://test-bucket/t1/part-00000-ec4adcbb-8616-4d93-ab3b-1343e49e2d9e-c000.snappy.parquet","partitionValues":{},"size":690,"modificationTime":1712605703000,"dataChange":true,"stats":"{\"numRecords\":4,\"minValues\":{\"a\":1,\"b\":11},\"maxValues\":{\"a\":4,\"b\":44},\"nullCount\":{\"a\":0,\"b\":0},\"tightBounds\":false}","deletionVector":{"storageType":"p","pathOrInlineDv":"s3://test-bucket/t1/deletion_vector_125d3fdb-d4c8-484d-938a-dfcc6aecbaae.bin","offset":1,"sizeInBytes":36,"cardinality":2}}}
```

> there should be a simpler test, not involving SHALLOW CLONE

My assumption is that storagetType `p` is specifically thought to handle shallow clone use cases so the current test needs `SHALLOW CLONE` to test the behavior.

Reply:
ok, then -- _there should be onTrino read from the baseTableName before the clone is created._